editor/editor_node.cpp

```diff
@@ -2161,6 +2161,8 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 
 			if (!p_confirmed) {
 				tab_closing = p_option == FILE_CLOSE ? editor_data.get_edited_scene() : _next_unsaved_scene(false);
+				_scene_tab_changed(tab_closing);
+				_update_scene_tabs();
```

Comment:
What is this code for? I removed it and it still works fine.

Reply:
This is more of a quality of life improvement. 
Say you have two unsaved scenes open (_tab1*_, _tab2*_) and you're focused on the tab to the right (_tab2*_). When you exit the editor, the popup asks you if you want to save. Since Godot scans the unsaved scenes from left to right, the popup will ask what you want to do with _tab1*_, but it's still focused on _tab2*_. You save/discard, _tab1_ gets closed and then reopened by this fix, but it's appended on the right. So now you have _tab2*_ and _tab1_, popup asks for _tab2*_ but the editor is focused on _tab1_.
These lines make the editor focus on the tab the popup is asking you what to do with, otherwise Godot will always focus on the latest one to the right, so the user can't really see the tab Godot is asking him to do something with, and it's impossible to select it when the popup is present.
I thought it's better to let the user see the tab, so that he can better see if it's to be saved or not.
Actually, I checked the code and it's pointless to call `_update_scene_tabs()`, because it will get called anyway if necessary, but `_scene_tab_changed` has to be called to fix the issue I've just explained.

Reply:
Ok the `_scene_tab_changed()` call is ok, but if `_update_scene_tabs()` is unnecessary then remove it.

Reply:
Ok, I applied everything and squashed the commits in a single one. It should be fine now